File: C#/raindrops/Raindrops.cs
using System;
using System.Text;

public static class Raindrops
{
    public static string Convert(int number)
    {
        StringBuilder sb = new StringBuilder();
        bool factor3 = number % 3 == 0;
        bool factor5 = number % 5 == 0;
        bool factor7 = number % 7 == 0;
        bool factor357 = factor3 || factor5 || factor7;

        if (factor3)
        {
            sb.Append("Pling");
        }
        if (factor5)
        {
            sb.Append("Plang");
        }
        if (factor7)
        {
            sb.Append("Plong");
        }
        if (!factor357)
        {
            sb.Append(number.ToString());
        }

        return sb.ToString();
    }
}

File: JavaScript/raindrops.js
//
// This is only a SKELETON file for the 'Raindrops' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const convert = (number) => {
  let factor3 = number % 3 == 0;
  let factor5 = number % 5 == 0;
  let factor7 = number % 7 == 0;
  let factor357 = factor3 || factor5 || factor7;
  let result = '';

  if (factor3) {
    result += 'Pling';
  }
  if (factor5) {
    result += 'Plang';
  }
  if (factor7) {
    result += 'Plong';
  }
  if (!factor357) {
    result += number;
  }
  return result;
};
